scripts-available/CDB_CartodbfyTable.sql

```diff
@@ -912,12 +928,25 @@ BEGIN
   -- indexes are in place and apply a rename
   SELECT *
   FROM _CDB_Has_Usable_Geom(reloid) 
-  AS (has_usable_geoms boolean, 
-      text_geom_column boolean, 
-      text_geom_column_name text, 
-      text_geom_column_srid boolean)
   INTO STRICT gc;
 
+  -- If geom is the wrong name, just rename it.
+  IF gc.has_geom AND gc.has_geom_name != const.geomcol THEN
+    sql := Format('ALTER TABLE %s DROP COLUMN IF EXISTS %I', reloid::text, const.geomcol);
+    PERFORM _CDB_SQL(sql,'_CDB_Rewrite_Table');
+    sql := Format('ALTER TABLE %s RENAME COLUMN %I TO %I', reloid::text, gc.has_geom_name, const.geomcol);
+    PERFORM _CDB_SQL(sql,'_CDB_Rewrite_Table');
+  END IF;
```

Comment:
Both drops were added in order to pass the new test. The renames were originally placed at the end of _CDB_Has_Usable_Geom

I double-checked the code and the prior renames are limited to dealing with cartodb_id column.

In practice I found this useful in cases where the source table has the_geom in a text type and some other such as wkb_geom with an actual geom that only needs renaming. Arguably we should rename instead of dropping.